src/interactive/Dropdown.jsx

```diff
@@ -71,27 +113,38 @@ class Dropdown extends React.PureComponent {
 			ref => !ref.contains(e.target)
 		);
 
-		if (isNotDropdownClick) {
+		if (isNotDropdownClick || (!isNotDropdownClick && this.props.closeOnBodyClick)) {
 			this.closeContent(e);
 		}
-	}
+	};
 
-	onBodyKeyDown(e) {
+	onBodyKeyDown = (e: SyntheticKeyboardEvent<*>) => {
 		if (e.key === 'Escape') {
-			this.closeContent();
+			this.closeContent(e);
 		}
-	}
+	};
 
 	componentDidMount() {
-		document.body.addEventListener('click', this.onBodyClick);
-		document.body.addEventListener('keydown', this.onBodyKeyDown);
+		if (document.body) {
+			document.body.addEventListener('click', this.onBodyClick);
+		}
+		if (document.body) {
```

Comment:
no need for the separate ifs :)

Reply:
Flow is actually so wierd about it:
![screen shot 2018-09-27 at 2 14 29 pm](https://user-images.githubusercontent.com/41023107/46171074-8564ce00-c26d-11e8-8124-94e7550c6cc1.png)

Reply:
ah, amazing, I see. Flow can't guarantee that calling `addEventListener` doesn't change the value of `document`, which is why it errors... hmmm I guess leave the two checks for now

Reply:
Just was thinking about this a little more, and it's _possible_ that you wouldn't get the Flow error if you used `window.document.body` rather than `document.body` because I think Flow treats `window` as immutable, so it _might_ not worry about `document` getting overwritten... really not sure though. Just wanted to leave the comment :)

Reply:
you can just use window in this case